### PR TITLE
chapter 30 created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
* [AP - Chapter 30] Custom views and touch events - https://androidtraining.atlassian.net/browse/AT-101

## Description:
Chapter 30 created

## :camera_flash: UI preview
ezgif com-video-to-gif(22)

### CODE REVIEW

File: app/src/main/res/layout/activity_drag_and_draw.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.bignerdranch.android.draganddraw.BoxDrawingView
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    />
```
Comment (f7deleon) [inline app/src/main/res/layout/activity_drag_and_draw.xml]: nl

**Review by f7deleon — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -16,4 +20,5 @@
             </intent-filter>
         </activity>
     </application>
-</manifest>
+
+</manifest>
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml]: nl

File: app/src/main/java/com/bignerdranch/android/draganddraw/Box.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.bignerdranch.android.draganddraw
+
+import android.graphics.PointF
+
+class Box(val start: PointF) {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/draganddraw/Box.kt:5]: you can use a RectF

File: app/src/main/java/com/bignerdranch/android/draganddraw/BoxDrawingView.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import android.graphics.PointF
+import android.util.AttributeSet
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+
+class BoxDrawingView(context: Context, attrs: AttributeSet? = null) : View(context, attrs) {
+  companion object {
+    private const val TAG = "BowDrawingView"
+    private const val BOX_COLOR = 0x22ff0000
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/draganddraw/BoxDrawingView.kt]: add it in the resource file.

File: app/src/main/java/com/bignerdranch/android/draganddraw/BoxDrawingView.kt [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    private const val BACKGROUND_COLOR = 0xfff8efe0
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/draganddraw/BoxDrawingView.kt]: same here

File: app/src/main/java/com/bignerdranch/android/draganddraw/DragAndDrawActivity.kt [hunk 6]
```diff
@@ -1,5 +1,12 @@
 package com.bignerdranch.android.draganddraw
 
 import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
 
-class DragAndDrawActivity : AppCompatActivity()
+class DragAndDrawActivity : AppCompatActivity() {
+
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/draganddraw/DragAndDrawActivity.kt]:
```suggestion
```

**Pull request merged**